packages/integrations/vercel/src/serverless/middleware.ts

```diff
@@ -0,0 +1,59 @@
+import { fileURLToPath, pathToFileURL } from 'node:url';
+import esbuild from 'esbuild';
+import { join } from 'node:path';
+import { writeFile } from '../lib/fs.js';
+
+/**
+ * It generates the Vercel Edge Middleware file.
+ *
+ * It creates a temporary file, the edge middleware, with some dynamic info.
+ *
+ * Then this file gets bundled with esbuild. The bundle phase will inline the Astro middleware code.
+ *
+ * @param astroMiddlewareEntryPoint
+ * @param outPath
+ * @returns {Promise<URL>} The path to the bundled file
+ */
+export async function generateEdgeMiddleware(
+	astroMiddlewareEntryPointPath: string,
+	outPath: string
+): Promise<URL> {
+	const code = edgeMiddlewareTemplate(astroMiddlewareEntryPointPath);
+	// Temporary file, so `esbuild` can resolve it without too much trouble.
+	// Using `stdin` option is not very optimal
+	const middlewareEdgePath = join(outPath, 'middleware-temp.mjs');
+	await writeFile(middlewareEdgePath, code);
```

Comment:
I'm curious why `stdin` isn't optimal? It should be fine for small amount of code passed 🤔 I'm using `stdin` here too:

https://github.com/withastro/astro/blob/1a9b644e20b9ff1e8484aae3e97a9ee2fa81238c/packages/astro/src/core/client-directive/build.ts#L12-L21